### PR TITLE
Allow DiffUtil to calculate on background thread

### PR DESCRIPTION
**Pull request by AleksanderMielczarek**
`FastAdapterDiffUtil` contains new method `calculateDiff` to allow calculate diff on background thread and later dispatch result to recycler on main thread using `set.`

Change is backward compatible.
Example included.

### CODE REVIEW

**Comment by FabianTerhorst** [comment]
Can you rebase to the feature/v3_new branch?

File: app/src/main/java/com/mikepenz/fastadapter/app/DiffUtilActivity.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+            @Override
+            public List<SimpleItem> call() throws Exception {
+                return createData();
+            }
+        }).map(new Function<List<SimpleItem>, DiffUtil.DiffResult>() {
+            @Override
+            public DiffUtil.DiffResult apply(List<SimpleItem> simpleItems) throws Exception {
+                return FastAdapterDiffUtil.calculateDiff(fastItemAdapter, simpleItems);
+            }
+        }).subscribeOn(Schedulers.newThread())
```
Comment (FabianTerhorst) [inline app/src/main/java/com/mikepenz/fastadapter/app/DiffUtilActivity.java]: Better use the thread pool here: Schedulers.io()

**Comment by AleksanderMielczarek** [comment]
Done.

**Comment by FabianTerhorst** [comment]
You have to target v3_new as well

**Comment by FabianTerhorst** [comment]
Not sure if fastItemAdapter should be explicitly supported here as well, since its almost unneeded in the next release and could probably get deleted.

**Comment by mikepenz** [comment]
@AleksanderMielczarek thank you very much. looks good. I'll have a look soon

**Comment by mikepenz** [comment]
@AleksanderMielczarek works amazing :) thank you very much

**Pull request merged**